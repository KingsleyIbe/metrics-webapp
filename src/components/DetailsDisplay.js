import { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { getData } from '../redux/home/home';

const DetailsDisplay = () => {
  const detailsList = useSelector((state) => state.covidReducer);

  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(getData());
  }, []);

  let Country;
  let slug;
  let newConfirmed;
  let totalConfirmed;
  let newDeaths;
  let totalDeaths;
  let newRecovered;
  let totalRecovered;
  let date;

  const fetchInfo = detailsList.map((countries) => {
    if (countries.CountryCode === window.location.pathname.slice(1)) {
      Country = countries.Country;
      date = countries.Date;
      slug = countries.Slug;
      newConfirmed = countries.NewConfirmed;
      totalConfirmed = countries.TotalConfirmed;
      newDeaths = countries.NewDeaths;
      totalDeaths = countries.TotalDeaths;
      newRecovered = countries.NewConfirmed;
      totalRecovered = countries.TotalRecovered;
      totalConfirmed = countries.TotalConfirmed;
    }
    return 'Data not found';
  });

  return (
    { fetchInfo },
      <div className="main-c">
        <div className="details-country" />
        <div className="name">
          <h2>{Country}</h2>
        </div>
        <div>
          <span className="updated-data">
            <span className="title">Statistics</span>
          </span>
          <div className="data">
            <span>Updated at:</span>
            <span>{date}</span>
          </div>
          <div className="data">
            <span>Confirmed New Cases:</span>
            <span>{newConfirmed}</span>
          </div>
          <div>
            <span className="title">Latest Data</span>
            <div className="data">
              <span>Total Confirmed:</span>
              <span>{totalConfirmed}</span>
            </div>
            <div className="data">
              <span>New Deaths:</span>
              <span>{newDeaths}</span>
            </div>
            <div className="data">
              <span>Slug:</span>
              <span>{slug}</span>
            </div>
          </div>
          <div>
            <span className="title">Recovered</span>
            <div className="data">
              <span>New Recovered:</span>
              <span>{newRecovered}</span>
            </div>
            <div className="data">
              <span>Total Recovered:</span>
              <span>{totalRecovered}</span>
            </div>
            <div className="data">
              <span>Total Deaths:</span>
              <span>{totalDeaths}</span>
            </div>
          </div>
        </div>
      </div>
  );
};

export default DetailsDisplay;
