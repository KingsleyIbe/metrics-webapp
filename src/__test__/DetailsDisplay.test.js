import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../Redux/configureStore';
import DetailsDisplay from '../components/DetailsDisplay';

describe('Render components', () => {
  test('render Details page', () => {
    const tree = renderer
      .create(<Router><Provider store={store}><DetailsDisplay /></Provider></Router>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
