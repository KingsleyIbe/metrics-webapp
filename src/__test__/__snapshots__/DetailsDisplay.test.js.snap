// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render components render Details page 1`] = `
<div
  className="main-c"
>
  <div
    className="details-country"
  />
  <div
    className="name"
  >
    <h2 />
  </div>
  <div>
    <span
      className="updated-data"
    >
      <span
        className="title"
      >
        Statistics
      </span>
    </span>
    <div
      className="data"
    >
      <span>
        Updated at:
      </span>
      <span />
    </div>
    <div
      className="data"
    >
      <span>
        Population:
      </span>
      <span />
    </div>
    <div>
      <span
        className="title"
      >
        Latest Data
      </span>
      <div
        className="data"
      >
        <span>
          Deaths:
        </span>
        <span />
      </div>
      <div
        className="data"
      >
        <span>
          Confirmed Cases:
        </span>
        <span />
      </div>
      <div
        className="data"
      >
        <span>
          Critical:
        </span>
        <span />
      </div>
    </div>
    <div>
      <span
        className="title"
      >
        Calculated
      </span>
      <div
        className="data"
      >
        <span>
          Death_Rate:
        </span>
        <span />
      </div>
      <div
        className="data"
      >
        <span>
          Recovery_Rate:
        </span>
        <span />
      </div>
      <div
        className="data"
      >
        <span>
          Recovered_vs_death_ratio:
        </span>
        <span />
      </div>
      <div
        className="data"
      >
        <span>
          Cases_per_million_population:
        </span>
        <span />
      </div>
    </div>
  </div>
</div>
`;
